humility-core/src/reflect.rs

```diff
@@ -807,6 +824,33 @@ pub fn load_enum(
     Ok(Enum(var.name.to_string(), val))
 }
 
+/// Loads a union from memory image `buf` at offset `addr`.
+///
+/// We assume that the only union types in our image are using
+/// `core::mem::MaybeUnint`, and extract the "value" variant.
```

Comment:
Maybe add an `assert!(ty.name.contains("MaybeUninit"), "only MaybeUninit unions supported")` as a support to this comment?